### ISSUE TITLE
[a11y] navigation rail is missing role

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#161542]
When using this widget https://api.flutter.dev/flutter/material/NavigationRail-class.html,  the semantics role should have the following

| OS | role |
|--------|--------|
| web | navigation |
| ios | - |
| macos |  - |
| windows | - |
| android | NavigationRailView | 

both web and android are missing

**Comment by chunhtai** [comment]
This mapping is inaccurate. navigation in web is referring to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav and NavigationRailView doesn't point to a specific role in the talkback source code as well. closing
